nibabel/viewers.py

```diff
@@ -0,0 +1,473 @@
+""" Utilities for viewing images
+
+Includes version of OrthoSlicer3D code originally written by our own
+Paul Ivanov.
+"""
+from __future__ import division, print_function
+
+import numpy as np
+import weakref
+
+from .optpkg import optional_package
+from .orientations import aff2axcodes, axcodes2ornt
+
+
+class OrthoSlicer3D(object):
+    """Orthogonal-plane slicer.
+
+    OrthoSlicer3d expects 3-dimensional data, and by default it creates a
+    figure with 3 axes, one for each slice orientation.
+
+    Clicking and dragging the mouse in any one axis will select out the
+    corresponding slices in the other two. Scrolling up and
+    down moves the slice up and down in the current axis.
+
+    OrthoSlicer3d also supports 4-dimensional data, where multiple
+    3-dimensional volumes are stacked along the last axis.  For 4-dimensional
+    data the fourth figure axis can be used to control which 3-dimensional
+    volume is displayed.  Alternatively, the - key can be used to decrement the
+    displayed volume and the + or = keys can be used to increment it.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> a = np.sin(np.linspace(0,np.pi,20))
+    >>> b = np.sin(np.linspace(0,np.pi*5,20))
+    >>> data = np.outer(a,b)[..., np.newaxis]*a
+    >>> OrthoSlicer3D(data).show()  # doctest: +SKIP
+    """
+    # Skip doctest above b/c not all systems have mpl installed
+    def __init__(self, data, affine=None, axes=None, cmap='gray',
```

Comment:
Any way of breaking this init up into smaller pieces?   It's very long.   If we find some smaller pieces, it may well make it easier to inherit from and modify.